[Canteen] Executable doesn't work with asdf-java on macOS

This issue is a little bit more exotic, I guess.

I'm using [asdf](https://asdf-vm.com/) and [asdf-java](https://github.com/halcyon/asdf-java) as version manager for managing the JRE/JDK environments on my machine running macOS.

Since the introduction of the executables created with Canteen in #188, I'm unable to use `reactive-grpc` with Gradle because `asdf` cannot find the configured JDK due to missing environment variables.

The generated executable seems to call the `java` command without any environment variables set, so that the wrapper script of `asdf` cannot use the `ASDF_DATA_DIR` or `HOME` environment variables to find the installed and configured JDKs.

FWIW, other protoc plugins seem to work (tested with this highly scientific method):
```
# /path/to/com.salesforce.servicelibs/reactor-grpc/1.0.1/8f96b6c394ef35e3dd520452cdb296c860f05d41/reactor-grpc-1.0.1-osx-x86_64.exe
unknown command: java. Perhaps you have to reshim?

# /path/to/io.grpc/protoc-gen-grpc-java/1.31.1/f28727b6793ace7c8797bd1ad41bb0bf9ece463d/protoc-gen-grpc-java-1.31.1-osx-x86_64.exe
[waiting for input]
^C
```

The wrapper script of `asdf` (shim) looks very simple and is generated via https://github.com/asdf-vm/asdf/blob/v0.8.0/lib/commands/command-reshim.bash#L76-L80, for example:
```bash
#!/usr/bin/env bash
# asdf-plugin: java adoptopenjdk-11.0.8+10
exec /root/.asdf/bin/asdf exec "java" "$@"
```

## Example

Full example project: [asdf-shim-error.zip](https://github.com/salesforce/reactive-grpc/files/5210528/asdf-shim-error.zip)

### build.gradle

```
plugins {
  id 'java-library'
  id 'com.google.protobuf' version '0.8.13'
}

repositories {
  jcenter()
}

dependencies {
  api 'io.grpc:grpc-protobuf:1.31.1'
  api 'io.grpc:grpc-stub:1.31.1'
  api 'io.projectreactor:reactor-core:3.3.9.RELEASE'
  implementation 'com.salesforce.servicelibs:reactor-grpc-stub:1.0.1'
}

protobuf {
  protoc {
    artifact = 'com.google.protobuf:protoc:3.13.0'
  }
  plugins {
    grpc {
      artifact = 'io.grpc:protoc-gen-grpc-java:1.31.1'
    }
    reactor { artifact = 'com.salesforce.servicelibs:reactor-grpc:1.0.1' }

  }
  generateProtoTasks {
    ofSourceSet('main')*.plugins {
      grpc {}
      reactor {}
    }
  }
}
```
### Error message
```
# ./gradlew generateProto
> Task :generateProto FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':generateProto'.
> protoc: stdout: . stderr: unknown command: java. Perhaps you have to reshim?
  --reactor_out: protoc-gen-reactor: Plugin failed with status code 1.


* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
3 actionable tasks: 1 executed, 2 up-to-date
```

Debug output (`./gradlew build --debug`): [debug.txt](https://github.com/salesforce/reactive-grpc/files/5210643/debug.txt)

--- a/reactor/reactor-grpc/pom.xml
+++ b/reactor/reactor-grpc/pom.xml
@@ -69,7 +69,7 @@
             <plugin>
                 <groupId>com.salesforce.servicelibs</groupId>
                 <artifactId>canteen-maven-plugin</artifactId>
-                <version>1.0.0</version>
+                <version>1.0.1</version>
                 <executions>
                     <execution>
                         <goals>

--- a/rx-java/rxgrpc/pom.xml
+++ b/rx-java/rxgrpc/pom.xml
@@ -69,7 +69,7 @@
             <plugin>
                 <groupId>com.salesforce.servicelibs</groupId>
                 <artifactId>canteen-maven-plugin</artifactId>
-                <version>1.0.0</version>
+                <version>1.0.1</version>
                 <executions>
                     <execution>
                         <goals>

